### DISCUSSION TITLE
Parity with Arc on Pinned Tab

### DISCUSSION THREAD

**Discussion by artcens** [zen-browser/desktop#2829] [General]
I'm really getting used to Arc, so having it match is awesome!
Since bookmarks are ugly in Zen, making pinned tabs act like bookmarks is great, and that's how pinned tabs in Arc work too. So the first thing to change is:
* Pinned Tab should not retain my last activity when exiting. 
[Resolved on #1957]
(it have even more customization!)

That alone can make Pinned Tab act more like a bookmark.
The list below is more changes to consider to make it even more perfect!
1. Ability to rename pinned tab or tab in general
[Resolved on #4979]

2. Ability to pin a tab by just dragging the tab to pinned or essential.
[Resolved on #4784]

3. Ability to group pinned tab or tab in general (in Arc, it's a folder)

4. Sidebar item synchronization: switching devices with less pain! ([idea!](https://github.com/zen-browser/desktop/discussions/2829#discussioncomment-11504504), originally from @erutan)

* from @chenseanxy 
[Resolved on #2904]
>  Resetting the URL for a pinned page should be a one-click operation too, not just through context menu and keyboard shortcuts

* from @erutan 
[Resolved 🟣 ??? - some say it was not]
> Add an option to display all pinned tabs without a scrollbar

* [reddit page](https://www.reddit.com/r/zen_browser/comments/1hrsb9u/isnt_there_a_way_to_close_pinned_tabs_without/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button)
[Resolved on 1.7.5b 🟣 - couldn't track the PR]
>I like how Arc handles this feature. "-" to close pinned tabs and "X" to remove pinned tabs. 

----
the experience is close to Arc in no time, same or better flow and behavior already!
If you have any suggestions to add, feel free to discuss! (I'll add them to the list!)

Thank you for taking your time to read this!

**Comment by chenseanxy** [comment]
> Pinned Tab should not retain my last activity when exit

This should already be available in version 19 I think

Just to add to the list of features:
- Resetting URL for a pinned page should be a one-click operation too, not just through context menu and keyboard shortcuts

**Reply by artcens** [reply]
> This should already be available in version 19 I think

if you sure it was a thing already then its either broken or i did something wrong, because its not resetting

**Reply by chenseanxy** [reply]
Yeah just double checked, links are not resetting despite `Restore pinned tabs to their originally pinned URL on startup` selected

**Comment by Futuremappermydud** [comment]
These features are definitely needed, currently pinned tabs are essentially useless as there is not much distinction from regular tabs.

**Comment by artcens** [comment]
does sidebar item(pinned and essential tab) get sync too or not? i assumes it's not, right? never try it before, can anyone confirm this?

**Reply by zacaronii** [reply]
They don't appear to. I just downloaded the twilight build alongside the normal alpha build. My workspaces did sync over but not my pins.

**Comment by erutan** [comment]
**Being able to rename Pinned Tab** 

This basically blocks me. Off the top of my head - having local/dev/test/live pinned tabs is really nice. I can figure it out by their relative placement, but that's non-ideal. Having to make them bookmarks to pin makes sense from a technical side. Perhaps just pinning a tab creates it as a bookmark in a (hidden?) folder.

**Being able to pin a page by just dragging the tab to the pinned tab section**

Yeah, it's painful without this.

I'll add a feature request:

**Add an option to display all pinned tabs without a scrollbar**

It makes sense to have this as an option past 6 pinned tabs if you're on a smaller screen.  Running with 1329 vertical points (2234 pixels) I can have over a dozen pinned tabs (for various staging sites, common resources for the project, etc) in a workspace and still have room for temporary ones below.

Overall I really like how zippy the browser is, and it'd be great to assign containers to workspaces!

**Reply by KiKaraage** [reply]
Rename Tab itself would be an important feature to have yeah.

**Reply by artcens** [reply]
> Perhaps just pinning a tab creates it as a bookmark in a (hidden?) folder.

That's a good idea, now pinned tabs or even essential tabs can get synchronized between devices. With some configuration, I'm sure they could make it work.
don't really need to be hidden, just name it "Pinned Tab" or "Sidebar Profile," since essential could get synced too.

I'm thinking bookmark setup like this:

#Sidebar Profile (Main Folder)
 | Essential Tabs (Folder)
 | Pinned Tabs (Folder)
 
Zen will search for "#Sidebar Profile" bookmark first. if it exist, then fetch the URL and the name of the pinned tab from there and put it on Sidebar.
yes, this is a hacky way to do it, but hey ... you got free cloud sync from Firefox account services xd

**Reply by erutan** [reply]
> don't really need to be hidden, just name it "Pinned Tab" or "Sidebar Profile," since essential could get synced too.

With Arc workspaces I don't use bookmarks, but I could imagine someone that wanted to use both having a bookmark folder that's redundant annoyed. It should probably be an option to toggle.

> yes, this is a hacky way to do it, but hey ... you got free cloud sync from Firefox account services xd

That's more or less what I was thinking. Two birds, one stone, less re-inventing the wheel.

**Comment by lucaphill** [comment]
One feature that would be nice to add is to automatically open a new tab when you close all your unpinned tabs instead of opening your first pinned tab.

https://github.com/user-attachments/assets/277643bc-536b-45e6-aaab-638bc6458276

**Reply by artcens** [reply]
isn't this already solved with the new zen URL bar? but rather open new tab, it just blank. and i personally like that.

**Reply by artcens** [reply]
wait, there's annoying behavior where if you still have normal tab when you unload all the pinned tab, it switch to that normal tab rather than blank page.

**Comment by nithinbull** [comment]
There are a couple of default behaviours in ARCs search bar missed by the original poster.

1. Ability to search for renamed tabs. 

Example: Let's say I have renamed by outlook pinned tab to "letter". The search results in the Arc search will show me the pinned tabs by default with their new name.

Default behaviour in Arc is as shown below, this is non existent in the Zen search, which lists open tabs based on their original title. I think the pinned tab renaming does not re-title them?

https://github.com/user-attachments/assets/e82e6036-15f4-46ca-b0f6-5dedd9de33ce


2. Pinned tabs on top of the search result. 

The Arc browser always focuses on the first result for return, in this case the first item will be the last opened tab. This enables workflows like cmd+t -> Return for returning to previous tab. And easily switching between the tabs with their new nick names.

In case of zen, I also have to enable the switch-to-tab functionality by typing "% + space" and on top of that I'll have to use the arrow keys to select the search result. And also the pinned tabs don't appear on the search results with their new name, as mentioned in the first point.

Arc :

https://github.com/user-attachments/assets/342581e8-8b3d-4dd0-aeff-40c7bb984fe4

**Comment by nithinbull** [comment]
Hey guys, it has been more than a year. But, there's no updates on this discussion.

**Reply by mr-cheffy** [reply]
There should be pretty much parity at this point.
